### PR TITLE
Feature/state pattern renderer

### PR DESCRIPTION
**Pull request by txsmith**

### CODE REVIEW

**Comment by txsmith** [comment]
The game and all game states can now attach event listening lambda's.

File: desktop/src/main/nl/tudelft/ti2806/riverrush/desktop/MainDesktop.java [hunk 1]
```diff
@@ -40,6 +39,13 @@ public MainDesktop() throws URISyntaxException {
 
         setupGraphics();
         client.connect();
+
+        try {
+            Thread.sleep(5000);
```
Comment (JobEngel) [inline desktop/src/main/nl/tudelft/ti2806/riverrush/desktop/MainDesktop.java]: Temporary Solution, right?

File: desktop/src/main/nl/tudelft/ti2806/riverrush/game/FinishedGameState.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package nl.tudelft.ti2806.riverrush.game;
```
Comment (JobEngel) [inline desktop/src/main/nl/tudelft/ti2806/riverrush/game/FinishedGameState.java]: Why are these classes considered new?

File: desktop/src/main/nl/tudelft/ti2806/riverrush/screen/FinishedGameScreen.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package nl.tudelft.ti2806.riverrush.screen;
+
```
Comment (JobEngel) [inline desktop/src/main/nl/tudelft/ti2806/riverrush/screen/FinishedGameScreen.java]: Probably should make this do something

**Pull request merged**